File: backend/app/schemas/data_update.py
from datetime import date

from pydantic import BaseModel


class SymbolUpdateSummaryResponse(BaseModel):
    symbol: str
    fetched_rows: int
    valid_rows: int
    changed_rows: int
    latest_date: date | None
    indicators_recalculated: bool
    indicator_rows: int
    status: str
    message: str


class MarketDataUpdateResponse(BaseModel):
    status: str
    latest_date: date | None
    processed_price_rows: int
    symbols_updated: list[str]
    indicators_recalculated: bool
    message: str
    details: list[SymbolUpdateSummaryResponse]

File: frontend/src/types/dataUpdate.ts
export interface SymbolUpdateSummary {
  symbol: string;
  fetched_rows: number;
  valid_rows: number;
  changed_rows: number;
  latest_date: string | null;
  indicators_recalculated: boolean;
  indicator_rows: number;
  status: string;
  message: string;
}

export interface MarketDataUpdateResponse {
  status: string;
  latest_date: string | null;
  processed_price_rows: number;
  symbols_updated: string[];
  indicators_recalculated: boolean;
  message: string;
  details: SymbolUpdateSummary[];
}
